presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowFlightClientHandler.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.plugin.arrow;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.ConnectorSession;
+import org.apache.arrow.flight.FlightClient;
+import org.apache.arrow.flight.FlightDescriptor;
+import org.apache.arrow.flight.FlightInfo;
+import org.apache.arrow.flight.Location;
+import org.apache.arrow.flight.grpc.CredentialCallOption;
+import org.apache.arrow.memory.RootAllocator;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.TimerTask;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.facebook.plugin.arrow.ArrowErrorCode.ARROW_FLIGHT_ERROR;
+
+public abstract class ArrowFlightClientHandler
+{
+    private static final Logger logger = Logger.get(ArrowFlightClientHandler.class);
+    private AtomicBoolean isClientClosed = new AtomicBoolean(true);
+    private final ArrowFlightConfig config;
+    private ScheduledExecutorService scheduledExecutorService;
+    private ArrowFlightClient arrowFlightClient;
+    private Optional<InputStream> trustedCertificate = Optional.empty();
+    private TimerTask closeTask;
+    private static final int TIMER_DURATION_IN_MINUTES = 30;
+
+    public ArrowFlightClientHandler(ArrowFlightConfig config)
+    {
+        this.config = config;
+    }
+
+    private void initializeClient(Optional<String> uri)
+    {
+        if (!isClientClosed.get()) {
+            return;
+        }
+        try {
+            RootAllocator allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
You would usually create a RootAllocator as a class member and they should be closed when not used anymore

Reply:
@BryanCutler I have made changes for closing allocator

Reply:
We are not re-using the flight client. The root allocator will be closed when `ArrowFlightClient` is closed or auto closed.

Reply:
where is this RootAllocator closed then? The FlightClient creates a child allocator that is closed along with the client, it will not close the root

Reply:
<img width="667" alt="image" src="https://github.com/user-attachments/assets/0007e8a2-0212-45ef-84d6-1917afe90a27">
<img width="546" alt="image" src="https://github.com/user-attachments/assets/616ecee8-5063-45d3-9039-902f71967d9e">

`ArrowFlightClient` saves the root allocator. The root allocator is closed when `ArrowFlightClient` is closed.

Reply:
You wouldn't normally create a RootAllocator for each client, it could be reused as each client will internally make a child allocator. It's ok to clean that up later, since you are closing it here.

Reply:
How would we determine when the root allocator should be closed? Should we keep track of open flight clients and close root allocator when there are no flight clients open for some duration of time?

Reply:
I would create 1 RootAllocator for the connector and then close it during `Connector.shutdown()`. The FlightClient creates it's own child allocator that is closed with the client, so the root won't be keeping any buffers directly.